components/status_dropdown/status_dropdown.tsx

```diff
@@ -24,76 +23,81 @@ import StatusOfflineIcon from 'components/widgets/icons/status_offline_icon';
+
 import './status_dropdown.scss';
 
-export default class StatusDropdown extends React.PureComponent {
-    static propTypes = {
-        style: PropTypes.object,
-        status: PropTypes.string,
-        userId: PropTypes.string.isRequired,
-        profilePicture: PropTypes.string,
-        autoResetPref: PropTypes.string,
-        actions: PropTypes.shape({
-            openModal: PropTypes.func.isRequired,
-            setStatus: PropTypes.func.isRequired,
-            unsetCustomStatus: PropTypes.func.isRequired,
-            setStatusDropdown: PropTypes.func.isRequired,
-        }).isRequired,
-        customStatus: PropTypes.object,
-        isCustomStatusEnabled: PropTypes.bool.isRequired,
-        isStatusDropdownOpen: PropTypes.bool.isRequired,
-        showCustomStatusPulsatingDot: PropTypes.bool.isRequired,
-    }
+type Props = {
+    status?: string;
+    userId: string;
+    profilePicture: string;
+    autoResetPref?: string;
+    actions: {
+        openModal: (modalData: {modalId: string; dialogType: any; dialogProps?: any}) => void;
+        setStatus: (status: UserStatus) => ActionFunc;
+        unsetCustomStatus: () => ActionFunc;
+        setStatusDropdown: (open: boolean) => void;
+    };
+    customStatus: UserCustomStatus | Record<string, any>;
+    isCustomStatusEnabled: boolean;
+    isStatusDropdownOpen: boolean;
+    showCustomStatusPulsatingDot: boolean;
+}
 
+export default class StatusDropdown extends React.PureComponent <Props> {
     static defaultProps = {
         userId: '',
         profilePicture: '',
         status: UserStatuses.OFFLINE,
         customStatus: {},
     }
 
-    isUserOutOfOffice = () => {
+    isUserOutOfOffice = (): boolean => {
         return this.props.status === UserStatuses.OUT_OF_OFFICE;
     }
 
-    setStatus = (status) => {
+    setStatus = (status: string): void => {
         this.props.actions.setStatus({
             user_id: this.props.userId,
             status,
+            manual: false,
+            last_activity_at: new Date().getTime(),
         });
     }
 
-    setOnline = (event) => {
+    setOnline = (event: Event): void => {
         event.preventDefault();
         this.setStatus(UserStatuses.ONLINE);
     }
 
-    setOffline = (event) => {
+    setOffline = (event: Event): void => {
         event.preventDefault();
         this.setStatus(UserStatuses.OFFLINE);
     }
 
-    setAway = (event) => {
+    setAway = (event: Event): void => {
         event.preventDefault();
         this.setStatus(UserStatuses.AWAY);
     }
 
-    setDnd = (event) => {
+    setDnd = (event: Event): void => {
         event.preventDefault();
         this.setStatus(UserStatuses.DND);
     }
 
-    showStatusChangeConfirmation = (status) => {
+    showStatusChangeConfirmation = (status: string): void => {
         const resetStatusModalData = {
-            ModalId: ModalIdentifiers.RESET_STATUS,
+            modalId: ModalIdentifiers.RESET_STATUS,
```

Comment:
Is the rename of `ModalId` to `modalId` supported by the code that expects this action?

Reply:
Hi @lieut-data  , I change it because after I check it on "actions/view/modals.js" the code is like :

```
import {ActionTypes} from 'utils/constants';

export function openModal(modalData) {
    return (dispatch) => {
        const action = {
            type: ActionTypes.MODAL_OPEN,
            modalId: modalData.modalId,
            dialogProps: modalData.dialogProps,
            dialogType: modalData.dialogType,
        };

        dispatch(action);
        return {data: true};
    };
}

// another function closeModal
```

As you can see,  in action. modalId, it receive modalData.modalId, not modalData.ModalId